### PR TITLE
Diagram "chatting with the cat"

### PR DESCRIPTION
**Pull request by sambarza**
First version of the not yet completed diagram of the chat process.
There is a new submenu "Chatting with the Cat" under menu "How the Cat works".

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "cSpell.words": [
+        "Embedder",
+        "pluggable"
+    ]
+}
```
Comment (nicola-corbellini) [inline .vscode/settings.json:6]: I think this should not be committed and pushed

**Review by nicola-corbellini — CHANGES_REQUESTED** [review]
Having a clickable diagram is a dream coming true

File: .vscode/settings.json (hunk 1 above)
Comment (sambarza) [reply .vscode/settings.json]: I agree, ignored the .vscode settings folder

**Review by nicola-corbellini — APPROVED** [review]

**Comment by nicola-corbellini** [comment]
Thank you very much!

**Pull request merged**